### DISCUSSION TITLE
Hacktoberfest and a `CONTRIBUTING.md`

### DISCUSSION THREAD

**Discussion by walidmujahid** [ledgersmb/LedgerSMB#8423] [Ideas]
I want to suggest that LedgerSMB participate in Hacktoberfest: https://hacktoberfest.com/participation/#maintainers

And connected to this suggestion is preparing a `CONTRIBUTING.md` file: https://mozillascience.github.io/working-open-workshop/contributing/

Even if this project doesn't find a need to participate in Hacktoberfest, I think it should still have a `CONTRIBUTING.md` file. I was recently directed to the development docs on the drupal site: https://github.com/ledgersmb/LedgerSMB/pull/8422#issuecomment-2394514195. That's great, but I think having a contributing file would be helpful in general for those that look for it (such as myself). Alternatively, or additionally, a section could be added to the README that links to the dev docs on the drupal site; call it `Development Guides` or something like that.

**Comment by neilt** [comment]
Regarding CONTRIBUTING.md and from matrix comment, I generally agree with 
[walidmujahid](https://matrix.to/#/@walidmujahid:matrix.org)
 's comments. The wiki articles are good for people searching the web and open to anyone to contribute, but when a developer wants to develop, it is much easier to consult a CONTRIBUTING.md file in the distribution that points to everything needed or important. As we have found, practically no one, except ehuelsmann, contributes to the wiki.  With the point that "anyone contributing" does not really apply.

When I started with LSMB, I found the wiki content difficult because the organization did not make sense to me. Not saying it was bad, but just that I had to search for everything, and was not able to just go to the source of truth. Some stuff was in the developer docker images, some in the wiki, some in tree itself, some in the book, etc. In some cases the wiki was out of date or conflicted with the repository files.

Also with the wiki there is no official way to review, comment, and approve content. For example I disagree with Chris's wiki entry about code comments, but since it is in the wiki, what is the correct way to get this reviewed and changed? I don't feel confortable just changing the wiki, especially one authored by a core contributer without some discussion. If the comment was in CONTRIBUTING.md I could submit a PR that would generate discussion and approval or not for the change.

In summary I think the wiki is great for less formal information, but I think important information should be in tree, under revision control, with approval ability. To me the decision about wiki vs in tree goes to how we assure the information is correct. Do we need PRs and approvals to change it, or is it really ok to just change based on anyone's opinion.

**Reply by ehuelsmann** [reply]
I've added the topic and issue labels. Please feel free to copy the site and wiki content into a proposal for the CONTRIBTING.md file so we can discuss and enter hacktoberfest

**Reply by ehuelsmann** [reply]
@neilt does the CONTRIBUTING.md file on master address this discussion? If so, can we close it?
